### PR TITLE
feat: port kafka native metrics to Prometheus 

### LINKED ISSUE #11187: feat(kafka source): support rdkafka statistics
### Is your feature request related to a problem? Please describe.

rdkafka can export statistics, which will be super helpful when debugging problems on user's side.

https://github.com/confluentinc/librdkafka/blob/master/STATISTICS.md

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tabVersion**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

related #11187

this pr only enables kafka native metrics in source part, sink part still needs some refactor and will enable in the next pr

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Comment by tabVersion** [comment]
@fuyufjh you can pick some metrics and I will port them to the dashboard

**Comment by fuyufjh** [comment]
> @fuyufjh you can pick some metrics and I will port them to the dashboard

My initial motivation is to observe the fetchq length in order to diagnose OOM problem, but some other metrics such as throughput etc. look helpful as well. I don't have lots of expertise on rdkafka so I can't give more suggestions. Please decide by your own 🙌

File: src/connector/src/sink/kafka.rs
Comment (fuyufjh) [inline src/connector/src/sink/kafka.rs:560]: What does this mean?👀

**Review by fuyufjh — APPROVED** [review]
Generally LGTM

File: src/connector/src/sink/kafka.rs
Comment (tabVersion) [reply src/connector/src/sink/kafka.rs:560]: `stats` serves as a callback func registered in kafka config context, we need to make matrics part of the context.

**Comment by tabVersion** [comment]
blocked by madsim-rdkafka, cc @wangrunji0408

**Comment by wangrunji0408** [comment]
> blocked by madsim-rdkafka, cc @wangrunji0408

I've updated madsim-rdkafka in #11706. Should be unblocked after it is merged.

**Pull request merged**
